Src/GoogleApis.Auth/OAuth2/UserCredential.cs

```diff
@@ -133,6 +124,23 @@ public void Initialize(ConfigurableHttpClient httpClient)
 
         #endregion
 
+        #region ITokenAccess implementation
+
+        public virtual async Task<string> GetAccessTokenForRequestAsync(string authUri = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (Token.IsExpired(flow.Clock))

```

Comment:
Above you protected the check on IsExpired with a null check, maybe add one here too if it can be null?